cluster/cluster.go

```diff
@@ -289,6 +290,91 @@ func CreateDeploymentWithTenants(tenants []*StorageGroupTenant, sg *StorageGroup
 	return err
 }
 
+// DeprovisionTenantOnStorageGroup deletes the tenant from the storage group and deletes all tenant's data from disks
+func DeprovisionTenantOnStorageGroup(ctx *Context, tenant *Tenant, sg *StorageGroup) chan error {
+	ch := make(chan error)
+	go func() {
+		defer close(ch)
+		if tenant == nil || sg == nil {
+			ch <- errors.New("nil Tenant or StorageGroup passed")
+			return
+		}
+
+		sgTenantResult := <-GetTenantStorageGroupByShortName(ctx, tenant.ShortName)
+		if sgTenantResult.Error != nil {
+			ch <- sgTenantResult.Error
+			return
+		}
+		ctx.Tenant = sgTenantResult.StorageGroupTenant.Tenant
+		// Get the credentials for a tenant
+		tenantConf, err := GetTenantConfig(tenant)
+		if err != nil {
+			ch <- err
+			return
+		}
+		// Unset Postgres notification before deleting tenant
+		err = unsetMinioConfigPostgresNotification(sgTenantResult.StorageGroupTenant, tenantConf)
+		if err != nil {
+			log.Println(err)
+			ch <- err
+			return
+		}
+		// Wait after minios restart
+		ready := isMinioReadyRetry(ctx)
```

Comment:
I think this is an unnecessary delay

Reply:
If we don't unset the bucket notification configuration, we can just keep going with the deprovisioning